Automattic/amp-wp
# Inherit object-fit and object-position styles so more conversions work out of the box

## Summary

This is a~n experimental~ follow-up on #5927. We can fix issues with `object-fit`/`object-position` for the Cover Block and for any other `amp-img`/`amp-video` that gets converted from `img`/`video` if we merely inherit the properties.

* Reduce specificity for style rule that applies `object-fit:contain` on images converted by the plugin. This is necessary for images which get a 100% width in content to prevent them from getting stretched or cropped. For example:

```html
<p><img src="https://amp.dev/static/img/icons/icon-72x72.png"></p>
```

Non-AMP | AMP
----------|------
![image](https://user-images.githubusercontent.com/134745/113364422-49ac9280-9308-11eb-81a3-976ff8dc00c6.png) | ![image](https://user-images.githubusercontent.com/134745/113364407-41ecee00-9308-11eb-8a94-6a713ee65fa1.png)

(Note the image does appear to be centered in AMP, but this is an existing issue: #4735.)

* Make sure that `object-fit`, `object-position`, and `image-resolution` are inherited from an AMP component wrapper to the contained shadow element and `noscript` fallback element. This allows us to entirely remove the `ampify_cover_block` logic.
* Eliminate obsolete CSS from core theme sanitizer:
   * Twenty Seventeen: 175,662 - 173,835 = 1,827 bytes saved (~1% before tree shaking).
   * Twenty Nineteen: 308,611 -  307,721 = 890 bytes saved (~1% before tree shaking).
* Omit attributes from `noscript` fallback elements which are likely to cause styling problems. Fixes #6028.

### QA

* Header image appears the same before/after in Twenty Nineteen.
* Header image and header video appear the same before/after in Twenty Seventeen.
* Cover block with image or video and assigned focal point appears the same as non-AMP.
* Image with `object-fit`, `object-position`, and `image-rendering` styles apply on the `amp-img > img` and `amp-img > noscript > img`, both with JavaScript enabled and disabled. For example, given a Custom HTML block with this markup: 

```html
<img 
    src="https://amp.dev/static/img/icons/icon-72x72.png"
    style="
        width: 300px;
        height: 100px;
        object-fit: cover;
        object-position: 100% 100%;
        image-rendering: pixelated;
    "
>
```

![image](https://user-images.githubusercontent.com/134745/113358263-52966780-92fa-11eb-8b9c-8683effaef14.png)

* Verify that `noscript` fallback elements are styled as expected. For example, a Custom HTML block with:

```html
<div style="outline:solid 1px black;">
<style>
.auto-height {
    object-fit: contain;
    object-position: 50% 50%;
    margin: auto;
    height: auto;
    -webkit-filter: grayscale(50%);
    filter: grayscale(50%); 
}
</style>
<img class="auto-height" src="https://amp.dev/static/img/icons/icon-128x128.png">
</div>
```

Non-AMP and AMP sans JS w/ fix 👍  | AMP Before with JS disabled 👎 
---------------|------------------------------
![image](https://user-images.githubusercontent.com/134745/113361475-f3882100-9300-11eb-89ee-ecac1f8c8f82.png) | ![image](https://user-images.githubusercontent.com/134745/113361662-4feb4080-9301-11eb-870d-91be42826eb2.png) 

## Checklist
